### ISSUE TITLE
feature request: fall back to use default list/launch/preview if not specified in custom provider

### ISSUE THREAD

**Issue by bdefore** [Biont/sway-launcher-desktop#51]
I'd like to have my launcher focus on an already opened instance of an application rather than open a new one, leveraging this library: https://gitlab.com/wef/dotfiles/-/blob/master/bin/sway-run-or-raise. Stretch goal: shift-enter overrides this and opens a new instance anyway (useful for say terminals)

I've authored the following custom provider in ~/.config/sway-launcher-desktop/providers.conf:

When i run `sway-launcher-desktop` I see what looks like the launcher is broken. Output of 0/0 and nothing I type matches. From what I gather in this conversation https://github.com/Biont/sway-launcher-desktop/issues/41 all three items are required to be specified in provider configs. But it's not clear to me what I should specify in a provider if I want default behavior.

Currently it looks like https://github.com/Biont/sway-launcher-desktop/blob/master/sway-launcher-desktop.sh#L35-L37 bails on setting up the provider unless all three are provided: https://github.com/Biont/sway-launcher-desktop/blob/master/sway-launcher-desktop.sh#L35-L37 ... perhaps this could be modified to fallback to default behavior per-item instead?

**Comment by Biont** [comment]
Yeah, the structure of a custom provider is described in the [readme](https://github.com/Biont/sway-launcher-desktop#extending-the-launcher) and all 3 need to be present. Fallbacks or inheritance are difficult since as you have notices, the script currently ignores the builtin providers if custom ones are specified.

I actually think what would work for you is a way to execute `run-or-raise` instead of [`setsid /bin/sh -c`](https://github.com/Biont/sway-launcher-desktop/blob/3c1c56380a82116e7f22a395e2b7a3febbbecd18/sway-launcher-desktop.sh#L323). This would catch *any* provider. Maybe a new environment variable could be introduced for this.

The other way would in fact be to make your provider complete, which would probably look similar to this:
(I really just pieced this together in the browser to give you an idea. I'd be surprised if it actually works, but it should help you get there)

```
[xdg-run-or-raise]
list_cmd=$HOME/sway-launcher-desktop.sh list-entries
preview_cmd=$HOME/sway-launcher-desktop.sh describe-desktop {1}
launch_cmd=$HOME/.config/sway-launcher-desktop/scripts/sway-run-or-raise $($HOME/sway-launcher-desktop.sh generate-command {1} {2})
```

Does this help you?
